include/itkTileMerging.hxx

```diff
@@ -0,0 +1,590 @@
+  //      oIt.Set(val);
+  //      ++oIt;
+  //      }
+  //    }
+  //  this->UpdateProgress((i + 1) / float(m_Regions.size()));
+  //  }
+  //return;
+
+  //now we will do resampling, one region at a time (in parallel)
+  //within each of these regions the set of contributing tiles is the same
+  MultiThreaderBase::Pointer mt = MultiThreaderBase::New();
+  MultiThreaderBase::ArrayThreadingFunctorType tf
+      = std::bind(&Self::ResampleSingleRegion, this, std::placeholders::_1);
+  mt->ParallelizeArray(0, m_Regions.size(), tf, this);
+
+  //release data from input tiles
+  RegionType reg0;
+  for (SizeValueType i = 0; i < this->m_LinearMontageSize; i++)
+    {
+    m_Metadata[i]->SetBufferedRegion(reg0);
+    m_Metadata[i]->Allocate(false);
+    }
+}
+
+template <typename TImageType, typename TInterpolator>
+void
+TileMerging<TImageType, TInterpolator>
+::ResampleSingleRegion(SizeValueType i)
+{
+  ImagePointer outputImage = this->GetOutput();
+  RegionType reqR = outputImage->GetRequestedRegion();
+  RegionType currentRegion = m_Regions[i];
+  if (!currentRegion.Crop(reqR)) //empty intersection
+    {
+    return; //nothing to do
+    }
+  bool interpolate = true;
+  if (this->m_PCMOptimizer->GetPeakInterpolationMethod() ==
+        Superclass::PCMOptimizerType::PeakInterpolationMethod::None)
+    {
+    interpolate = false;
+    }
+  ImageRegionIteratorWithIndex<ImageType> oIt(outputImage, currentRegion);
+
+  if (m_RegionContributors[i].empty()) //not covered by any tile
+    {
+    while (!oIt.IsAtEnd())
+      {
+      oIt.Set(m_Background);
+      ++oIt;
+      }
+    }
+  else if (m_RegionContributors[i].size() == 1) //just one tile
+    {
+    SizeValueType tileIndex = *m_RegionContributors[i].begin();
+    TileIndexType nDIndex = this->LinearIndexTonDIndex(tileIndex);
+    OffsetType tileToRegion = m_Regions[i].GetIndex() - m_InputMappings[tileIndex].GetIndex();
+    RegionType iReg = currentRegion;
+    iReg.SetIndex(m_Metadata[tileIndex]->GetLargestPossibleRegion().GetIndex() + tileToRegion);
+    ImageConstPointer input = this->GetImage(nDIndex, iReg);
+
+    if (!interpolate)
+      {
+      ImageAlgorithm::Copy(input.GetPointer(), outputImage.GetPointer(), iReg, currentRegion);
+      }
+    else
+      {
+      Vector<typename ContinuousIndexType::ValueType, ImageDimension> continuousIndexDifference;
+      for (unsigned d = 0; d < ImageDimension; d++)
+        {
+        continuousIndexDifference[d] = input->GetLargestPossibleRegion().GetIndex(d)
+            - m_InputsContinuousIndices[tileIndex][d];
+        }
+      typename TInterpolator::Pointer interp = TInterpolator::New();
+      interp->SetInputImage(input);
+      while (!oIt.IsAtEnd())
+        {
+        ContinuousIndexType continuousIndex = oIt.GetIndex();
+        continuousIndex += continuousIndexDifference;
+        oIt.Set(interp->EvaluateAtContinuousIndex(continuousIndex));
+        ++oIt;
+        }
+      }
+    }
+  else //more than one tile contributes
```

Comment:
Does this work when the output image region is split for streaming?

Reply:
Yes